lib/pos_blockchain.go

```diff
@@ -368,20 +366,30 @@ func (bc *Blockchain) validateQC(desoBlock *MsgDeSoBlock, validatorSet []*Valida
 	return nil
 }
 
-// validateTimeoutQC validates that the parent block hash is not too far back from the latest.
-// Specifically, it checks that the parent block hash is at least the latest committed block.
-func (bc *Blockchain) validateTimeoutQC(desoBlock *MsgDeSoBlock, validatorSet []*ValidatorEntry) error {
-	// TODO: Implement me
-	return errors.New("IMPLEMENT ME")
-}
-
-// hasKnownAncestors checks that the parent block hash exists in the block index.
-// It returns true if the parent block hash exists in the block index, and otherwise false.
-// If the parent hash is in the block index, it is assumed that all ancestors of this block
-// also exist in the block index.
-func (bc *Blockchain) hasKnownAncestors(parentHash BlockHash) bool {
-	_, exists := bc.blockIndex[parentHash]
-	return exists
+// getAncestorsToCommittedTip returns the ancestors of the block provided up to the
+// committed tip.
+func (bc *Blockchain) getAncestorsToCommittedTip(desoBlock *MsgDeSoBlock) ([]*BlockNode, error) {
+	highestCommittedBlock, idx := bc.getHighestCommittedBlock()
+	if idx == -1 || highestCommittedBlock == nil {
+		return nil, errors.New("getAncestorsToCommittedTip: No committed blocks found")
+	}
+	currentHash := desoBlock.Header.PrevBlockHash.NewBlockHash()
+	ancestors := []*BlockNode{}
+	for {
+		currentBlock, exists := bc.blockIndex[*currentHash]
+		if !exists {
+			return nil, RuleErrorMissingAncestorBlock
+		}
+		ancestors = append(ancestors, currentBlock)
+		if currentBlock.Hash.IsEqual(highestCommittedBlock.Hash) {
+			break
+		}
+		if currentBlock.CommittedStatus == COMMITTED {
+			return nil, RuleErrorDoesNotExtendCommittedTip
+		}
+		currentHash = currentBlock.Header.PrevBlockHash
+	}
+	return ancestors, nil
```

Comment:
Consider reversing the order order of blocks in the output so the ancestors are sorted by block height in increasing order